debian/ubuntu-advantage-tools.postinst

```diff
@@ -95,45 +98,96 @@ if status:
+    apt_suite="${release}-${service}";
+    case "${service}" in
+        apps)
+            apt_origin="UbuntuESMApps"
+            apt_pref_file=${ESM_APPS_APT_PREF_FILE};
+            apt_source_file=${ESM_APPS_APT_SOURCE_FILE};
+            apt_key=${ESM_APPS_KEY};
+            ;;
+        infra)
+            apt_origin="UbuntuESM"
+            apt_pref_file=${ESM_INFRA_APT_PREF_FILE};
+            apt_source_file=${ESM_INFRA_APT_SOURCE_FILE};
+            apt_key=${ESM_INFRA_KEY_TRUSTY};
+            ;;
+    esac
+
+    # GPG key setup to avoid apt gpg key warnings
+    if [ ! -f "$APT_TRUSTED_KEY_DIR/$apt_key" ]; then
+        cp $UA_KEYRING_DIR/$apt_key $APT_TRUSTED_KEY_DIR
     fi
 
-    if [ -e "$ESM_APT_SOURCE_FILE_TRUSTY" ]; then
-        mv $ESM_APT_SOURCE_FILE_TRUSTY $ESM_INFRA_APT_SOURCE_FILE_TRUSTY
-    fi
-    if [ -e "$ESM_APT_PREF_FILE_TRUSTY" ]; then
-        mv $ESM_APT_PREF_FILE_TRUSTY $ESM_INFRA_APT_PREF_FILE_TRUSTY
+    # Migrate trusty legacy source list and preference file names
+    if [ "14.04" = "$VERSION_ID" ]; then
+        if [ -e "$ESM_APT_SOURCE_FILE_TRUSTY" ]; then
+            mv $ESM_APT_SOURCE_FILE_TRUSTY $ESM_INFRA_APT_SOURCE_FILE
+        fi
+        if [ -e "$ESM_APT_PREF_FILE_TRUSTY" ]; then
+            mv "$ESM_APT_PREF_FILE_TRUSTY" "$ESM_INFRA_APT_PREF_FILE"
+        fi
     fi
-    if [ ! -e "$ESM_INFRA_APT_SOURCE_FILE_TRUSTY" ]; then
-        cat > $ESM_INFRA_APT_SOURCE_FILE_TRUSTY <<EOF
+    # If preference file doesn't already exist, we aren't attached.
+    # Setup unauthenticated apt source list file and never-pin preference
+    if [ ! -e "${apt_source_file}" ]; then
+        # Unconfigured repo, so set it up as never-pinned
+        cat > ${apt_source_file} <<EOF
 # Written by ubuntu-advantage-tools
-deb https://esm.ubuntu.com/ubuntu trusty-infra-security main
-# deb-src https://esm.ubuntu.com/ubuntu trusty-infra-security main
+deb https://esm.ubuntu.com/${service}/ubuntu ${apt_suite}-security main
+# deb-src https://esm.ubuntu.com/${service}/ubuntu ${apt_suite}-security main
 
-deb https://esm.ubuntu.com/ubuntu trusty-infra-updates main
-# deb-src https://esm.ubuntu.com/ubuntu trusty-infra-updates main
+deb https://esm.ubuntu.com/${service}/ubuntu ${apt_suite}-updates main
+# deb-src https://esm.ubuntu.com/${service}/ubuntu ${apt_suite}-updates main
 EOF
+
         # Automatically disable esm sources via apt preferences until enabled
-        cat > $ESM_INFRA_APT_PREF_FILE_TRUSTY <<EOF
+        cat > "${apt_pref_file}" <<EOF
 # Written by ubuntu-advantage-tools
 Package: *
-Pin: release o=UbuntuESM, n=trusty
+Pin: release o=${apt_origin}, n=${release}
 Pin-Priority: never
 EOF
     fi
+
+}
+
+configure_esm() {
+    rm -f $APT_TRUSTED_KEY_DIR/ubuntu-esm*gpg  # Remove legacy esm keys
+    case "${VERSION_ID}" in
+        14.04)
+            enable_apps="false" enable_infra="true" release="trusty" ;;
+        16.04)
+            enable_apps="false" enable_infra="true" release="xenial" ;;
```

Comment:
Shouldn't `enable_apps` here be true ?

Reply:
and yes, I had noted that when preparing the integration tests. Good catch